### PR TITLE
Mesos commons - add readable protobuf code

### PR DESCRIPTION
**Pull request by kensipe**
added a new project (which is intended to grow up and out on its own)
new project mesos-common has utilities which are commons mesos utilities and have 0 hdfs code
added significant code around protobuf.   There is more to add for generic capabilities, however it includes most of what we need for HDFS and tests.

### CODE REVIEW

**Comment by kensipe** [comment]
@gabrielhartmann looking for review.  I did some code clean up and added docs since we reviewed last night.

File: hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java [hunk 1]
```diff
@@ -16,13 +16,13 @@
 import org.apache.mesos.Protos.Status;
 import org.apache.mesos.Protos.TaskInfo;
 import org.apache.mesos.Protos.TaskState;
-import org.apache.mesos.Protos.TaskStatus;
+import org.apache.mesos.file.FileUtils;
 import org.apache.mesos.hdfs.config.HdfsFrameworkConfig;
 import org.apache.mesos.hdfs.config.NodeConfig;
```
Comment (gabrielhartmann) [inline hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java]: Is there some automated style checking thing available for consistent organization of imports?  Not requesting anything like that for this PR.  Just a thought that it'd be nice to get consistent import organization.

File: hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java [hunk 2]
```diff
@@ -105,6 +105,7 @@ private void setUpDataDir() {
    * Create Symbolic Link for the HDFS binary.
    */
   private void createSymbolicLink(ExecutorDriver driver) {
+    // todo:  (kgs) file util
```
Comment (gabrielhartmann) [inline hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java]: Count me as a vote against in code Tasks.  I'd prefer JIRAs so all issues are centralized in one place.  It also forces a clearer definition of what needs to be done.  What happens if you get hit by a bus Ken?  We will never know how to complete this important task.

File: hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java [hunk 3]
```diff
@@ -202,12 +200,11 @@ protected Process startProcess(ExecutorDriver driver, Task task) {
...
-        processBuilder.environment().putAll(createHdfsNodeEnvironment(task));
-        proc = processBuilder.start();
-        procWatcher.watch(proc);
-        task.setProcess(proc);
-        redirectProcess(task.getProcess());
+        Map<String, String> envMap = createHdfsNodeEnvironment(task);
+
+        Process process = ProcessUtil.startCmd(envMap, task.getCmd());
+        procWatcher.watch(process);
+        task.setProcess(process);
```
Comment (gabrielhartmann) [inline hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java]: This process in the "Task" stuff is bad.  Again, probably not something that needs to be addressed in this PR.

File: mesos-commons/src/main/java/org/apache/mesos/process/ProcessUtil.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+
+/**
+ * Helps to create processes for command lines  commonly used in mesos.
+ */
```
Comment (gabrielhartmann) [inline mesos-commons/src/main/java/org/apache/mesos/process/ProcessUtil.java]: extra space after "lines"

File: mesos-commons/src/main/java/org/apache/mesos/protobuf/EnvironmentBuilder.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+import org.apache.mesos.Protos;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Builder class for working with protobufs.  It includes 2 different approaches;
+ * 1) static functions useful for developers that want helpful protobuf functions for CommandInfo.
```
Comment (gabrielhartmann) [inline mesos-commons/src/main/java/org/apache/mesos/protobuf/EnvironmentBuilder.java]: "CommandInfo" shouldn't be here.  Comment needs updating.

File: mesos-commons/src/main/java/org/apache/mesos/protobuf/ExecutorInfoBuilder.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
+package org.apache.mesos.protobuf;
+
+import org.apache.mesos.Protos;
+
+import java.util.List;
+
+/**
+ * Builder class for working with protobufs.  It includes 2 different approaches;
+ * 1) static functions useful for developers that want helpful protobuf functions for CommandInfo.
```
Comment (gabrielhartmann) [inline mesos-commons/src/main/java/org/apache/mesos/protobuf/ExecutorInfoBuilder.java]: "CommandInfo" shouldn't be here.  Comment needs updating.

File: mesos-commons/src/main/java/org/apache/mesos/protobuf/LabelBuilder.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
+package org.apache.mesos.protobuf;
+
+import org.apache.mesos.Protos;
+
+/**
+ * Builder class for working with protobufs.  It includes 2 different approaches;
+ * 1) static functions useful for developers that want helpful protobuf functions for CommandInfo.
```
Comment (gabrielhartmann) [inline mesos-commons/src/main/java/org/apache/mesos/protobuf/LabelBuilder.java]: "CommandInfo" shouldn't be here.  Comment needs updating.

File: mesos-commons/src/main/java/org/apache/mesos/protobuf/OfferBuilder.java [hunk 8]
```diff
@@ -0,0 +1,94 @@
+package org.apache.mesos.protobuf;
+
+import org.apache.mesos.Protos;
+
+import java.util.List;
+
+/**
+ * Builder class for working with protobufs.  It includes 2 different approaches;
+ * 1) static functions useful for developers that want helpful protobuf functions for CommandInfo.
```
Comment (gabrielhartmann) [inline mesos-commons/src/main/java/org/apache/mesos/protobuf/OfferBuilder.java]: "CommandInfo" shouldn't be here.  Comment needs updating.

File: mesos-commons/src/main/java/org/apache/mesos/protobuf/ResourceBuilder.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
+package org.apache.mesos.protobuf;
+
+import org.apache.mesos.Protos.Resource;
+import org.apache.mesos.Protos.Value;
+
+/**
+ * Builder class for working with protobufs.  It includes 2 different approaches;
+ * 1) static functions useful for developers that want helpful protobuf functions for CommandInfo.
```
Comment (gabrielhartmann) [inline mesos-commons/src/main/java/org/apache/mesos/protobuf/ResourceBuilder.java]: "CommandInfo" shouldn't be here.  Comment needs updating.

File: mesos-commons/src/main/java/org/apache/mesos/protobuf/TaskInfoBuilder.java [hunk 10]
```diff
@@ -0,0 +1,72 @@
+package org.apache.mesos.protobuf;
+
+import com.google.protobuf.ByteString;
+import org.apache.mesos.Protos;
+
+import java.util.List;
+
+/**
+ * Builder class for working with protobufs.  It includes 2 different approaches;
+ * 1) static functions useful for developers that want helpful protobuf functions for CommandInfo.
```
Comment (gabrielhartmann) [inline mesos-commons/src/main/java/org/apache/mesos/protobuf/TaskInfoBuilder.java]: "CommandInfo" shouldn't be here.  Comment needs updating.

File: mesos-commons/src/main/java/org/apache/mesos/protobuf/TaskStatusBuilder.java [hunk 11]
```diff
@@ -0,0 +1,111 @@
+package org.apache.mesos.protobuf;
+
+import com.google.protobuf.ByteString;
+import org.apache.mesos.Protos;
+
+/**
+ * Builder class for working with protobufs.  It includes 2 different approaches;
+ * 1) static functions useful for developers that want helpful protobuf functions for CommandInfo.
```
Comment (gabrielhartmann) [inline mesos-commons/src/main/java/org/apache/mesos/protobuf/TaskStatusBuilder.java]: "CommandInfo" shouldn't be here.  Comment needs updating.

File: mesos-commons/src/main/java/org/apache/mesos/protobuf/TaskUtil.java [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+import org.apache.mesos.Protos;
+
+/**
+ * Utility class for working with Tasks.   This class reduces the overhead of protobuf and makes code
+ * easier to read.
+ */
+public class TaskUtil {
+
+
+
```
Comment (gabrielhartmann) [inline mesos-commons/src/main/java/org/apache/mesos/protobuf/TaskUtil.java]: Too many new lines.

**Comment by gabrielhartmann** [comment]
Squash commits.  Minor comments.  LGTM.

File: hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java (hunk 2 above)
Comment (kensipe) [inline hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java]: lol bus collision countermeasures activitied

File: hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java (hunk 1 above)
Comment (kensipe) [inline hdfs-executor/src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java]: IDEA does import organization and I almost always run that prior to check in.  If you're looking for gradle version... I imagine something like that exists.

**Comment by kensipe** [comment]
squash updated and tested on DCOS testing

**Pull request merged**